res/controllers/Numark-Mixtrack-Platinum-scripts.js

```diff
@@ -0,0 +1,1415 @@
+                this.stopFlash();
+            }
+        },
+        startFlash: function() {
+            // already flashing
+            if (this.flash_timer) {
+                engine.stopTimer(this.flash_timer);
+            }
+
+            this.flash_state = false;
+            this.send(this.on);
+
+            var time = 500;
+            if (this.inGetValue() > 0) {
+                time = 150;
+            }
+
+            var button = this;
+            this.flash_timer = engine.beginTimer(time, function() {
+                if (button.flash_state) {
+                    button.send(button.on);
+                    button.flash_state = false;
+                }
+                else {
+                    button.send(button.off);
+                    button.flash_state = true;
+                }
+            });
+        },
+        stopFlash: function() {
+            engine.stopTimer(this.flash_timer);
+            this.flash_timer = null;
+            this.trigger();
+        },
+    });
+
+    this.touch_strip = new this.EffectUnitTouchStrip();
+    this.enableButtons = new components.ComponentContainer();
+    for (var n = 1; n <= 3; n++) {
+        this.enableButtons[n] = new this.EffectEnableButton(n);
+    }
+
+    this.bpmTap = new this.BpmTapButton();
+
+    this.enableButtons.reconnectComponents();
+
+    this.forEachComponent(function (component) {
+        if (component.group === undefined) {
+            component.group = eu.group;
+        }
+    });
+};
+MixtrackPlatinum.EffectUnit.prototype = new components.ComponentContainer();
+
+MixtrackPlatinum.Deck = function(number, midi_chan, effects_unit) {
+    var deck = this;
+    var eu = effects_unit;
+    this.active = (number == 1 || number == 2);
+
+    components.Deck.call(this, number);
+    this.play_button = new components.PlayButton({
+        midi: [0x90 + midi_chan, 0x00],
+        off: 0x01,
+        sendShifted: true,
+        shiftControl: true,
+        shiftOffset: 4,
+        unshift: function() {
+            components.PlayButton.prototype.unshift.call(this);
+            this.type = components.Button.prototype.types.toggle;
+        },
+        shift: function() {
+            this.inKey = 'play_stutter';
+            this.type = components.Button.prototype.types.push;
+        },
+    });
+
+    this.load = new components.Button({
+        inKey: 'LoadSelectedTrack',
+        shift: function() {
+            if (ShiftLoadEjects) {
+                this.inKey = 'eject';
+            }
+            else {
+                this.inKey = 'LoadSelectedTrackAndPlay';
+            }
+        },
+        unshift: function() {
+            this.inKey = 'LoadSelectedTrack';
+        },
+    });
+
+    this.cue_button = new components.CueButton({
+        midi: [0x90 + midi_chan, 0x01],
+        off: 0x01,
+        sendShifted: true,
+        shiftControl: true,
+        shiftOffset: 4,
+        shift: function() {
+            this.inKey = 'start';
```

Comment:
Why not just use the default shift behavior of `components.CueButton`?

Reply:
When I look at the icon printed on the controller, I would expect the default `start_stop` behavior. I don't really understand the use case for not stopping playback with this. If you want to jump to the start and keep playing, you can use a normal hotcue.

> The reasoning here is that most people who get this controller and use it with mixxx won't be familar with mixxx.

I don't think that's a reliable assumption. Many Mixxx users do not have (much) experience with other DJ software, especially Linux users.

Reply:
I'm ok with going with the `start_stop` default. Thinking this through, jumping to the start and continuing to play doesn't seem very useful. Shift+play will jump to the cue point and I can't think of a reason you would want to jump to the start of the track instead of the cue point and keep playing. I always use this button to jump to the start when not playing the track, so making this the default would actually save me a button press.